Updates to net-tools sample test script

pfl opened the pull request:
A few updates to scripts/net/run-sample-tests.sh script enabling overlays for the Zephyr build process and running both UDP and TCP echo tests.

pfl committed:
scripts: net: Add overlay file to sample script

Add an '--overlay' command line option in order to use an overlay
file when compiling Zephyr.

Signed-off-by: Patrik Flykt <patrik.flykt@intel.com>

pfl committed:
scripts: net: Return final return value from Docker test

Return the final return value from Docker script when running echo
client in the container, first for UDP and then TCP, for both IPv4
and IPv6.

Signed-off-by: Patrik Flykt <patrik.flykt@intel.com>

pfl committed:
scripts: net: Add option to keep container after test

Add '--keep' option to leave Docker container and network running
after the test.

Signed-off-by: Patrik Flykt <patrik.flykt@intel.com>

jukkar commented on scripts/net/run-sample-tests.sh:
```diff
@@ -394,6 +404,17 @@ do
+
+	--overlay)
+	    shift
+	    if [ -n "$zephyr_overlay" ]
```
Can't we just say `zephyr_overlay="$zephyr_overlay $1"` here? Is there some shell that does not accept this?

pfl replied:
We get an extra space into that variable in such a case, but that should mean harm.

jukkar commented on scripts/net/run-sample-tests.sh:
```diff
@@ -164,7 +164,7 @@ start_zephyr ()
     fi
```
Why do we need / want to use 64-bit native_posix? That info would be nice to have in commit msg.

pfl replied:
Actually, we don't. Let's skip this patch.

pfl committed:
scripts: net: Enhance error reporting

Enhance error reporting by returning immediately when the calling
function reports an error.

Signed-off-by: Andrei Laperie <andrei.laperie@intel.com>
Signed-off-by: Patrik Flykt <patrik.flykt@intel.com>

jukkar reviewed (approved):
LGTM
